pkg/bid/bid.go

```diff
@@ -0,0 +1,149 @@
+package bid
+
+import (
+	"fmt"
+	"strings"
+
+	v2 "github.com/conductorone/baton-sdk/pb/c1/connector/v2"
+)
+
+const (
+	BidPrefix            = "bid:"
+	ResourceBidPrefix    = BidPrefix + "r:"
+	EntitlementBidPrefix = BidPrefix + "e:"
+	GrantBidPrefix       = BidPrefix + "g:"
+)
+
+func ToBid(b interface{}) string {
+	switch bType := b.(type) {
+	case *v2.Resource:
+		return ResourceToBid(bType)
+	case *v2.Entitlement:
+		return EntitlementToBid(bType)
+	case *v2.Grant:
+		return GrantToBid(bType)
+	}
+	return "☠️" // TODO: panic? this should never happen
```

Comment:
The pattern I've seen is 2 exported forms of this function:

'MakeBID' which returns an error, and 'MustMakeBID' which is a convenience wrapper that panics.